File: strings/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1

        bit_max = 2**31
        bit_min = -(2**31)

        rev = int(str(abs(x))[::-1]) * sign

        return rev if  bit_min <= rev <= (bit_max - 1) else 0

File: strings/reverse_integer.rb
# @param {Integer} x
# @return {Integer}
def reverse(x)
    result = 0
    if x.to_s.include?("-") 
        result = -x.to_s.reverse.to_i 
    else
        result = x.to_s.reverse.to_i
    end

    bit_max = 2**31
    bit_min = -2**31

    return 0 if result > bit_max || result < bit_min

    result
end